File: src/backend/Booking.Api/Domain/Entities/SleepingAccommodation.cs
using Booking.Api.Domain.Common;
using Booking.Api.Domain.Enums;

namespace Booking.Api.Domain.Entities;

public class SleepingAccommodation : IAuditableEntity, ISoftDeletable
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public required string Name { get; set; }
    public AccommodationType Type { get; set; }
    public int MaxCapacity { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ChangedAt { get; set; }
}

File: src/frontend/lib/types/sleeping-accommodation.ts
export enum AccommodationType {
  Room = 0,
  Tent = 1,
  Camper = 2,
  Other = 3
}

export interface SleepingAccommodation {
  id: string;
  name: string;
  type: AccommodationType;
  maxCapacity: number;
  isActive: boolean;
  createdAt: string;
  changedAt?: string;
}

export interface CreateSleepingAccommodationDto {
  name: string;
  type: AccommodationType;
  maxCapacity: number;
}

export interface UpdateSleepingAccommodationDto {
  name: string;
  type: AccommodationType;
  maxCapacity: number;
  isActive: boolean;
}